### DISCUSSION TITLE
Fresh Netbox & PostgreSQL installation

### DISCUSSION THREAD

**Discussion by Sharky21** [netbox-community/netbox#6556] [Q&A] [answered]
I'm brand new to Netbox installation/configuration so I apologise for DFU questions...

I want to new installation from scratch of the latest version of Netbox. However, I want to install the Netbox on ServerX with the PostgreSQL installed and running on ServerY. Both servers are running Ubuntu 18.04.

I've taken a look through the documentation and watched the installation video and it all seems fine but I've got a couple of questions if I may:

1. Is it ok to run (both fresh installations) Netbox and PostgreSQL on different servers (same network)? Is there any special configuration I need to do to allow the two services to communicate with each other?

2. If Netbox is installed on ServerX, is that where I need to install Redis, Gunicorn etc or any of these required to be running on the PostgresSQL (ServerY) host?

3. ServerX is already running the Apache2 http server. Is this ok to use with the fresh install of Netbox? Are there any instructions on how to configure Netbox with an existing Apache2 install?

Thanks for any advice...

**Comment by jeremystretch** [comment, marked as answer]
> Is it ok to run (both fresh installations) Netbox and PostgreSQL on different servers (same network)?

Yes; this is fairly common.

> Is there any special configuration I need to do to allow the two services to communicate with each other?

The PosgtreSQL server must be configured to accept and authenticate remote connections. Although not covered by the NetBox documentation, this is easy to do and detailed in the PostgreSQL documentation.

> If Netbox is installed on ServerX, is that where I need to install Redis, Gunicorn etc or any of these required to be running on the PostgresSQL (ServerY) host?

Redis, like PostgreSQL, may be installed either locally or on a remote host. (If installed on a remote host, is must also be configured appropriate to accept remote connections.) Gunicorn is what runs the Python worker processes; these run on the NetBox server.

> ServerX is already running the Apache2 http server. Is this ok to use with the fresh install of Netbox?

Generally, yes, but you'll need to first understand _why_ Apache2 is already running. E.g. is it serving a different application that might conflict with NetBox?

> Are there any instructions on how to configure Netbox with an existing Apache2 install?

The Apache2 documentation details how to serve multiple virtual servers; there's nothing special about NetBox in this regard. You can find a minimal Apache2 config example [here](https://github.com/netbox-community/netbox/blob/develop/contrib/apache.conf).

Hope that helps!

**Reply by Sharky21** [reply]
Thanks Jeremy, much appreciated. I'll give it a crack and see how it goes...
